blocks.py, models.py, custom_table.html

johnnyporkchops opened the pull request:
Created a custom table block so a content editor can add statistical press release tables and also use for creating other table and text sections as needed. Issue : 813
#813

This follows [this design](https://cloud.githubusercontent.com/assets/11636908/22669601/6148369e-ec93-11e6-9614-83656c2b5351.png) as closely as possible while also balancing the concern that:
1) The exact structure of these tables will vary
2) It should be useful for creating other similar content sections as needed
3) It will be an available block (within the ubiquitous streamfactory) on several templates and if content editors use it-- it should be specific enough to create the statistical press releases but not so rigid that it is cumbersome to use it for other table/text combined content use-cases.

It is setup as a streamblock with the option to add more that one table for each "custom_table section", allowing the ability to stack tables closely within one section or put them into separate horizontally- spaced sections. (They can even stack adjacent tables with no space between to achieve the "mid-table header" format that some of the legacy site tables seem to have, although @jenniferthibault's  [design ](https://cloud.githubusercontent.com/assets/11636908/22669601/6148369e-ec93-11e6-9614-83656c2b5351.png)departs from that, thankfully.)

Adding corresponding styles to:
/scss/elements/_tables.scss
https://github.com/18F/fec-style/pull/649

Screenshot of wagtail page:
![scrnsht](https://cloud.githubusercontent.com/assets/5572856/24090767/93b8d290-0d19-11e7-96b4-88a3e95afb61.jpg)

xtine commented on fec/home/blocks.py:
```diff
@@ -147,3 +147,25 @@ class DocumentFeedBlurb(blocks.StructBlock):
     description = blocks.CharBlock()
+
+class CustomTableBlock(blocks.StructBlock):
+    """A custom table"""
```
Can you expand on this comment, like for example what kind of pages or data this table might be used for.

johnnyporkchops replied:
Will expanp on comment

xtine commented on fec/home/templates/blocks/custom_table.html:
```diff
@@ -0,0 +1,23 @@
+  {% if block.block_type == "title" %}
+    <h4>{{ block }}</h4>
+  
+  {% elif block.block_type == "table" %}
```
Can you combine the `block.block_type` check for both `table` and `table_intro` into one line since they give the same output?

johnnyporkchops replied:
Sorry, I am not sure what you mean by the above comment, can you please elaborate?

xtine replied:
`{% elif block.blocktype == "title" or block.blocktype == "table" %}`

johnnyporkchops replied:
OK, now I see what you mean, they don't give the same output, they both use the same tag syntax to generate their output. Yes can I change that as now that I am fairly sure that I will that edits will not require wrapping either in an html block .

noahmanger commented on fec/home/blocks.py:
```diff
@@ -166,3 +187,24 @@ class ExampleForms(blocks.StructBlock):
         template = 'blocks/example-forms.html'
         icon = 'doc-empty'
+
+class CustomTableBlock(blocks.StructBlock):
```
Is this an accident having this in here twice?

johnnyporkchops replied:
Yes that is a mistake, I thought I was starting with a fresh checkout instead of a merge so i just went to the bottom and added by block no realizing that it was merged above the new block

johnnyporkchops commented:
**home/templates/blocks/custom_table.html**: Removed extraneous markup from this template by removing a hardcoded wrapper div that I had originally added--Wagtail appends the class ```block-custom_table``` to the body div and I use that as the wrapper class instead.

**home/models.py** had ``` ContactInfoBlock``` listed twice in the import statements at the top -- I removed that.

**home/blocks.py** : Removed the extra block definition for CustomTableBlock
